src/containers/Cinema/Cinema.tsx

```diff
@@ -1,192 +1,147 @@
 import { useWatchHistoryListener } from '#src/hooks/useWatchHistoryListener';
-import { useWatchHistoryStore } from '#src/stores/WatchHistoryStore';
-import { addScript } from '#src/utils/dom';
-import useOttAnalytics from '#src/hooks/useOttAnalytics';
-import { deepCopy } from '#src/utils/collection';
-import type { JWPlayer } from '#types/jwplayer';
 import type { PlaylistItem } from '#types/playlist';
 import { saveItem } from '#src/stores/WatchHistoryController';
 import { usePlaylistItemCallback } from '#src/hooks/usePlaylistItemCallback';
-import useEventCallback from '#src/hooks/useEventCallback';
 import { useConfigStore } from '#src/stores/ConfigStore';
+import Fade from '#src/components/Animation/Fade/Fade';
+import IconButton from '#src/components/IconButton/IconButton';
+import ArrowLeft from '#src/icons/ArrowLeft';
+import Player from '#src/components/Player/Player';
+import type { JWPlayer } from '#types/jwplayer';
+import { useWatchHistoryStore } from '#src/stores/WatchHistoryStore';
+import { VideoProgressMinMax } from '#src/config';
 
 type Props = {
+  open: boolean;
   item: PlaylistItem;
   onPlay?: () => void;
   onPause?: () => void;
   onComplete?: () => void;
-  onUserActive?: () => void;
-  onUserInActive?: () => void;
+  onClose?: () => void;
   feedId?: string;
-  isTrailer?: boolean;
-  playerId?: string;
+  title: string;
+  videoMeta: string;
+  isSeries: boolean;
+  seriesMeta?: string;
+  episodeTitle?: string;
 };
 
-const Cinema: React.FC<Props> = ({ item, onPlay, onPause, onComplete, onUserActive, onUserInActive, feedId, isTrailer = false }: Props) => {
+const Cinema: React.FC<Props> = ({ open, item, title, videoMeta, isSeries, seriesMeta, episodeTitle, onPlay, onPause, onComplete, onClose, feedId }: Props) => {
+  const { t } = useTranslation();
   const { player, features } = useConfigStore((s) => s.config);
   const continueWatchingList = features?.continueWatchingList;
+  const enableWatchHistory = !!continueWatchingList;
 
-  const playerElementRef = useRef<HTMLDivElement>(null);
-  const playerRef = useRef<JWPlayer>();
-  const loadingRef = useRef(false);
-  const seekToRef = useRef(-1);
-  const [libLoaded, setLibLoaded] = useState(!!window.jwplayer);
-  const scriptUrl = `${import.meta.env.APP_API_BASE_URL}/libraries/${player}.js`;
-  const enableWatchHistory = continueWatchingList && !isTrailer;
-  const setPlayer = useOttAnalytics(item, feedId);
-  const handlePlaylistItemCallback = usePlaylistItemCallback();
+  // state
+  const [playerInstance, setPlayerInstance] = useState<JWPlayer>();
+  const [isPlaying, setIsPlaying] = useState<boolean>(false);
+  const [userActive, setUserActive] = useState(true);
 
-  const getProgress = useCallback((): number | null => {
-    if (!playerRef.current) return null;
+  // watch history
+  const watchHistoryItem = useWatchHistoryStore((state) => item && state.getItem(item));
+  const videoProgress = watchHistoryItem?.progress;
```

Comment:
Should we still do this if `enableWatchHistory` is false?

Reply:
Are you suggesting something like this?

```tsx
  // watch history
  const watchHistoryItem = useWatchHistoryStore((state) => !!item && enableWatchHistory ? state.getItem(item) : undefined);

  const startTime = useMemo(() => {
    const videoProgress = watchHistoryItem?.progress;

    if (videoProgress && videoProgress > VideoProgressMinMax.Min && videoProgress < VideoProgressMinMax.Max) {
      return videoProgress * item.duration;
    }

    return 0;
  }, [item.duration, watchHistoryItem?.progress]);
```

Reply:
Yes, nice solution without breaking the hooks!